identity_stronghold/src/stronghold_jwk_storage_ext.rs

```diff
@@ -0,0 +1,158 @@
+    let key_type = StrongholdKeyType::try_from(&key_type)?;
+
+    if !matches!(key_type, StrongholdKeyType::BLS12381G2) {
+      return Err(
+        KeyStorageError::new(KeyStorageErrorKind::UnsupportedKeyType)
+          .with_custom_message(format!("{key_type} is not supported")),
+      );
+    }
+
+    let stronghold = self.get_stronghold().await;
+    let client = get_client(&stronghold)?;
+
+    let kid: KeyId = random_key_id();
+    let target_key_location = Location::generic(
+      IDENTITY_VAULT_PATH.as_bytes().to_vec(),
+      kid.to_string().as_bytes().to_vec(),
+    );
+    let jwk = client
+      .exec_proc([], &target_key_location, |_| {
+        let (sk, pk) = generate_bbs_keypair(alg).map_err(|e| FatalProcedureError::from(e.to_string()))?;
+        let public_jwk = encode_bls_jwk(&sk, &pk, alg).1;
+
+        Ok(Products {
+          output: public_jwk,
+          secret: Zeroizing::new(sk.to_bytes().to_vec()),
+        })
+      })
+      .map_err(|e| {
+        KeyStorageError::new(KeyStorageErrorKind::Unspecified)
+          .with_custom_message("Failed to execute stronghold procedure")
+          .with_source(e)
+      })?;
+
+    persist_changes(self, stronghold).await?;
+
+    Ok(JwkGenOutput::new(kid, jwk))
+  }
+
+  async fn sign_bbs(
+    &self,
+    key_id: &KeyId,
+    data: &[Vec<u8>],
+    header: &[u8],
+    public_key: &Jwk,
+  ) -> KeyStorageResult<Vec<u8>> {
+    let stronghold = self.get_stronghold().await;
+    let client = get_client(&stronghold)?;
+
+    // Extract the required alg from the given public key
+    let alg = public_key
+      .alg()
+      .ok_or(KeyStorageErrorKind::UnsupportedProofAlgorithm)
+      .and_then(|alg_str| {
+        ProofAlgorithm::from_str(alg_str).map_err(|_| KeyStorageErrorKind::UnsupportedProofAlgorithm)
+      })?;
+
+    // Check `key_id` exists in store.
+    if !self.exists(key_id).await? {
+      return Err(KeyStorageError::new(KeyStorageErrorKind::KeyNotFound));
+    }
+
+    let pk = expand_bls_jwk(public_key)
+      .map_err(|e| KeyStorageError::new(KeyStorageErrorKind::Unspecified).with_source(e))?
+      .1;
+
+    let sk_location = Location::Generic {
+      vault_path: IDENTITY_VAULT_PATH.as_bytes().to_vec(),
+      record_path: key_id.to_string().as_bytes().to_vec(),
+    };
+
+    client
+      .get_guards([sk_location], |[sk]| {
+        let sk = BBSplusSecretKey::from_bytes(&sk.borrow()).map_err(|e| FatalProcedureError::from(e.to_string()))?;
+        sign_bbs(alg, data, &sk, &pk, header).map_err(|e| FatalProcedureError::from(e.to_string()))
+      })
+      .map(|sig| sig.to_vec())
+      .map_err(|e| {
+        KeyStorageError::new(KeyStorageErrorKind::Unspecified)
+          .with_custom_message("Signature failed")
+          .with_source(e)
+      })
+  }
+
+  async fn update_signature(
+    &self,
+    key_id: &KeyId,
+    public_key: &Jwk,
+    signature: &[u8; BBSplusSignature::BYTES],
```

Comment:
Inconsistent API: `sign_bbs` returns signature as `Vec<u8>`, `update_signature` takes and returns signature as `[u8; N]`.